src/otx/algo/segmentation/huggingface_model.py

```diff
@@ -0,0 +1,145 @@
+    """
+
+    def __init__(
+        self,
+        model_name_or_path: str,  # https://huggingface.co/models?pipeline_tag=image-segmentation
+        label_info: LabelInfoTypes,
+        optimizer: OptimizerCallable = DefaultOptimizerCallable,
+        scheduler: LRSchedulerCallable | LRSchedulerListCallable = DefaultSchedulerCallable,
+        metric: MetricCallable = SegmCallable,  # type: ignore[assignment]
+        torch_compile: bool = False,
+    ) -> None:
+        self.model_name = model_name_or_path
+        self.load_from = None
+
+        super().__init__(
+            label_info=label_info,
+            optimizer=optimizer,
+            scheduler=scheduler,
+            metric=metric,
+            torch_compile=torch_compile,
+        )
+        self.image_processor = AutoImageProcessor.from_pretrained(self.model_name)
+
+    def _create_model(self) -> nn.Module:
+        return AutoModelForSemanticSegmentation.from_pretrained(
+            pretrained_model_name_or_path=self.model_name,
+            num_labels=self.label_info.num_classes,
+            ignore_mismatched_sizes=True,
+        )
+
+    def _customize_inputs(self, entity: SegBatchDataEntity) -> dict[str, Any]:
+        masks = torch.stack(entity.masks).long() if self.training else None
+
+        return {
+            "pixel_values": entity.images,
+            "labels": masks,
+        }
+
+    def _customize_outputs(
+        self,
+        outputs: SemanticSegmenterOutput,
+        inputs: SegBatchDataEntity,
+    ) -> SegBatchPredEntity | OTXBatchLossEntity:
+        if self.training:
+            return OTXBatchLossEntity(loss=outputs.loss)
+        if self.explain_mode:
+            msg = "Explain mode is not supported yet."
+            raise NotImplementedError(msg)
+
+        target_sizes = [info.img_shape for info in inputs.imgs_info]
+        results = self.image_processor.post_process_semantic_segmentation(
+            outputs,
+            target_sizes=target_sizes,
+        )
+
+        return SegBatchPredEntity(
+            batch_size=inputs.batch_size,
+            images=inputs.images,
+            imgs_info=inputs.imgs_info,
+            scores=[],
+            masks=results,
+        )
+
+    @property
+    def _exporter(self) -> OTXModelExporter:
+        """Creates OTXModelExporter object that can export the model."""
+        size = self.image_processor.size.values()
+        size = (*size, *size) if len(size) == 1 else size
+        image_size = (1, 3, *size)
+        image_mean = (123.675, 116.28, 103.53)
+        image_std = (58.395, 57.12, 57.375)
+
+        return OTXNativeModelExporter(
+            task_level_export_parameters=self._export_parameters,
+            input_size=image_size,
+            mean=image_mean,
+            std=image_std,
+            resize_mode="standard",
+            pad_value=0,
+            swap_rgb=False,
+            via_onnx=False,
+            onnx_export_configuration=None,
```

Comment:
Is it OK to leave ONNX config empty? Does it work?

Reply:
I haven't noticed any particular issues when I've tried it.